File: java/Samples/Biometrics/CS/AbisSampleCS/Settings/IrisesSettingsPage.cs
using System;
using System.Collections.Specialized;
using System.Windows.Forms;
using Neurotec.Biometrics;
using Neurotec.Devices;

namespace Neurotec.Samples
{
	public partial class IrisesSettingsPage : Neurotec.Samples.SettingsPageBase
	{
		#region Public constructor

		public IrisesSettingsPage()
		{
			InitializeComponent();

			cbMatchingSpeed.Items.Add(NMatchingSpeed.High);
			cbMatchingSpeed.Items.Add(NMatchingSpeed.Medium);
			cbMatchingSpeed.Items.Add(NMatchingSpeed.Low);

			cbTemplateSize.Items.Add(NTemplateSize.Large);
			cbTemplateSize.Items.Add(NTemplateSize.Medium);
			cbTemplateSize.Items.Add(NTemplateSize.Small);
		}

		#endregion

		#region Private methods

		private void ListDevices()
		{
			try
			{
				var current = Client.IrisScanner;
				var selected = cbScanners.SelectedItem;
				cbScanners.BeginUpdate();
				cbScanners.Items.Clear();
				foreach (NDevice item in Client.DeviceManager.Devices)
				{
					if ((item.DeviceType & NDeviceType.IrisScanner) == NDeviceType.IrisScanner)
					{
						cbScanners.Items.Add(item);
					}
				}

				cbScanners.SelectedItem = current;
				if (cbScanners.SelectedIndex == -1 && cbScanners.Items.Count > 0)
				{
					cbScanners.SelectedItem = selected;
					if (cbScanners.SelectedIndex == -1 && cbScanners.Items.Count > 0)
					{
						cbScanners.SelectedIndex = 0;
					}
				}
			}
			catch (Exception ex)
			{
				Utilities.ShowError(ex);
			}
			finally
			{
				cbScanners.EndUpdate();
			}
		}

		#endregion

		#region Public methods

		public override void OnNavigatingFrom()
		{
			Client.DeviceManager.Devices.CollectionChanged -= DevicesCollectionChanged;

			base.OnNavigatingFrom();
		}

		public override void OnNavigatedTo(params object[] args)
		{
			base.OnNavigatedTo(args);

			Client.DeviceManager.Devices.CollectionChanged += DevicesCollectionChanged;
		}

		public override void LoadSettings()
		{
			ListDevices();

			cbScanners.SelectedItem = Client.IrisScanner;
			cbTemplateSize.SelectedItem = Client.IrisesTemplateSize;
			cbMatchingSpeed.SelectedItem = Client.IrisesMatchingSpeed;
			nudMaximalRotation.Value = Convert.ToDecimal(Client.IrisesMaximalRotation);
			nudQuality.Value = Client.IrisesQualityThreshold;
			chbFastExtraction.Checked = Client.IrisesFastExtraction;

			base.LoadSettings();
		}

		public override void DefaultSettings()
		{
			if (cbScanners.Items.Count > 0)
				cbScanners.SelectedIndex = 0;
			Client.ResetProperty("Irises.TemplateSize");
			Client.ResetProperty("Irises.MatchingSpeed");
			Client.ResetProperty("Irises.MaximalRotation");
			Client.ResetProperty("Irises.QualityThreshold");
			Client.ResetProperty("Irises.FastExtraction");

			base.DefaultSettings();
		}

		#endregion

		#region Private events

		private void DevicesCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
		{
			BeginInvoke(new MethodInvoker(ListDevices));
		}

		private void CbScannersSelectedIndexChanged(object sender, EventArgs e)
		{
			Client.IrisScanner = cbScanners.SelectedItem as NIrisScanner;
		}

		private void CbTemplateSizeSelectedIndexChanged(object sender, EventArgs e)
		{
			Client.IrisesTemplateSize = (NTemplateSize)cbTemplateSize.SelectedItem;
		}

		private void CbMatchingSpeedSelectedIndexChanged(object sender, EventArgs e)
		{
			Client.IrisesMatchingSpeed = (NMatchingSpeed)cbMatchingSpeed.SelectedItem;
		}

		private void NudMaximalRotationValueChanged(object sender, EventArgs e)
		{
			Client.IrisesMaximalRotation = Convert.ToSingle(nudMaximalRotation.Value);
		}

		private void NudQualityValueChanged(object sender, EventArgs e)
		{
			Client.IrisesQualityThreshold = Convert.ToByte(nudQuality.Value);
		}

		private void ChbFastExtractionCheckedChanged(object sender, EventArgs e)
		{
			Client.IrisesFastExtraction = chbFastExtraction.Checked;
		}

		#endregion
	}
}

File: java/Samples/Biometrics/VB/AbisSampleVB/Settings/IrisesSettingsPage.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Specialized
Imports System.Windows.Forms
Imports Neurotec.Biometrics
Imports Neurotec.Devices

Partial Public Class IrisesSettingsPage
	Inherits Neurotec.Samples.SettingsPageBase
#Region "Public constructor"

	Public Sub New()
		InitializeComponent()

		cbMatchingSpeed.Items.Add(NMatchingSpeed.High)
		cbMatchingSpeed.Items.Add(NMatchingSpeed.Medium)
		cbMatchingSpeed.Items.Add(NMatchingSpeed.Low)

		cbTemplateSize.Items.Add(NTemplateSize.Large)
		cbTemplateSize.Items.Add(NTemplateSize.Medium)
		cbTemplateSize.Items.Add(NTemplateSize.Small)

		AddHandler cbScanners.SelectedIndexChanged, AddressOf CbScannersSelectedIndexChanged
		AddHandler cbTemplateSize.SelectedIndexChanged, AddressOf CbTemplateSizeSelectedIndexChanged
		AddHandler cbMatchingSpeed.SelectedIndexChanged, AddressOf CbMatchingSpeedSelectedIndexChanged
		AddHandler nudMaximalRotation.ValueChanged, AddressOf NudMaximalRotationValueChanged
		AddHandler nudQuality.ValueChanged, AddressOf NudQualityValueChanged
		AddHandler chbFastExtraction.CheckedChanged, AddressOf ChbFastExtractionCheckedChanged
	End Sub

#End Region

#Region "Private methods"

	Private Sub ListDevices()
		Try
			Dim current = Client.IrisScanner
			Dim selected = cbScanners.SelectedItem
			cbScanners.BeginUpdate()
			cbScanners.Items.Clear()
			For Each item As NDevice In Client.DeviceManager.Devices
				If (item.DeviceType And NDeviceType.IrisScanner) = NDeviceType.IrisScanner Then
					cbScanners.Items.Add(item)
				End If
			Next item

			cbScanners.SelectedItem = current
			If cbScanners.SelectedIndex = -1 AndAlso cbScanners.Items.Count > 0 Then
				cbScanners.SelectedItem = selected
				If cbScanners.SelectedIndex = -1 AndAlso cbScanners.Items.Count > 0 Then
					cbScanners.SelectedIndex = 0
				End If
			End If
		Catch ex As Exception
			Utilities.ShowError(ex)
		Finally
			cbScanners.EndUpdate()
		End Try
	End Sub

#End Region

#Region "Public methods"

	Public Overrides Sub OnNavigatingFrom()
		RemoveHandler Client.DeviceManager.Devices.CollectionChanged, AddressOf DevicesCollectionChanged

		MyBase.OnNavigatingFrom()
	End Sub

	Public Overrides Sub OnNavigatedTo(ByVal ParamArray args() As Object)
		MyBase.OnNavigatedTo(args)

		AddHandler Client.DeviceManager.Devices.CollectionChanged, AddressOf DevicesCollectionChanged
	End Sub

	Public Overrides Sub LoadSettings()
		ListDevices()

		cbScanners.SelectedItem = Client.IrisScanner
		cbTemplateSize.SelectedItem = Client.IrisesTemplateSize
		cbMatchingSpeed.SelectedItem = Client.IrisesMatchingSpeed
		nudMaximalRotation.Value = Convert.ToDecimal(Client.IrisesMaximalRotation)
		nudQuality.Value = Client.IrisesQualityThreshold
		chbFastExtraction.Checked = Client.IrisesFastExtraction

		MyBase.LoadSettings()
	End Sub

	Public Overrides Sub DefaultSettings()
		If cbScanners.Items.Count > 0 Then
			cbScanners.SelectedIndex = 0
		End If
		Client.ResetProperty("Irises.TemplateSize")
		Client.ResetProperty("Irises.MatchingSpeed")
		Client.ResetProperty("Irises.MaximalRotation")
		Client.ResetProperty("Irises.QualityThreshold")
		Client.ResetProperty("Irises.FastExtraction")

		MyBase.DefaultSettings()
	End Sub

#End Region

#Region "Private events"

	Private Sub DevicesCollectionChanged(ByVal sender As Object, ByVal e As NotifyCollectionChangedEventArgs)
		BeginInvoke(New MethodInvoker(AddressOf ListDevices))
	End Sub

	Private Sub CbScannersSelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)
		Client.IrisScanner = TryCast(cbScanners.SelectedItem, NIrisScanner)
	End Sub

	Private Sub CbTemplateSizeSelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)
		Client.IrisesTemplateSize = CType(cbTemplateSize.SelectedItem, NTemplateSize)
	End Sub

	Private Sub CbMatchingSpeedSelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)
		Client.IrisesMatchingSpeed = CType(cbMatchingSpeed.SelectedItem, NMatchingSpeed)
	End Sub

	Private Sub NudMaximalRotationValueChanged(ByVal sender As Object, ByVal e As EventArgs)
		Client.IrisesMaximalRotation = Convert.ToSingle(nudMaximalRotation.Value)
	End Sub

	Private Sub NudQualityValueChanged(ByVal sender As Object, ByVal e As EventArgs)
		Client.IrisesQualityThreshold = Convert.ToByte(nudQuality.Value)
	End Sub

	Private Sub ChbFastExtractionCheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
		Client.IrisesFastExtraction = chbFastExtraction.Checked
	End Sub

#End Region
End Class
